boot/assets.go

```diff
@@ -581,7 +581,8 @@ func (o *TrustedAssetsUpdateObserver) BeforeWrite() error {
 		// boot assets was updated
 		return nil
 	}
-	if err := resealKeyToModeenv(dirs.GlobalRootDir, o.model, o.modeenv); err != nil {
+	const expectReseal = true
+	if err := resealKeyToModeenv(dirs.GlobalRootDir, o.model, o.modeenv, expectReseal); err != nil {
```

Comment:
this is because we know an update is happening from the fact we are in `BeforeWrite()` and thus we necessarily must reseal against the asset we are updating correct?

Reply:
both BeforeWrite and Cancel starts with a guard like:

```
	if o.modeenv == nil {
		// modeenv wasn't even loaded yet, meaning none of the boot
		// assets was updated
		return nil
	}
```

so if we get this far is quite sure we modified the modeenv and we have a reason to reseal, also that notwithstanding setting expectReseal to true is the safer choice anyway

Reply:
ok, yeah that all makes sense then, thanks for the double confirmation